File: swift/file_splitter/file_splitter_18.swift
// Welcome to the File Splitter program!
// This delightful program is designed to read files and split their contents into smaller chunks.
// It is crafted with love and care to help you understand the intricacies of file handling in Swift.
// Let's embark on this colorful journey together!

import Foundation

// Function to read the contents of a file
func readFileContents(atPath path: String) -> String? {
    // Attempt to read the file at the given path
    do {
        let contents = try String(contentsOfFile: path, encoding: .utf8)
        return contents
    } catch {
        // If an error occurs, print a message and return nil
        print("Oh no! There was an error reading the file at path: \(path)")
        return nil
    }
}

// Function to split the contents of a file into smaller chunks
func splitContents(_ contents: String, chunkSize: Int) -> [String] {
    // Create an array to hold the chunks
    var chunks: [String] = []
    var startIndex = contents.startIndex
    
    // Loop through the contents and create chunks
    while startIndex < contents.endIndex {
        let endIndex = contents.index(startIndex, offsetBy: chunkSize, limitedBy: contents.endIndex) ?? contents.endIndex
        let chunk = String(contents[startIndex..<endIndex])
        chunks.append(chunk)
        startIndex = endIndex
    }
    
    return chunks
}

// Function to write chunks to separate files
func writeChunks(_ chunks: [String], toDirectory directory: String) {
    // Loop through the chunks and write each one to a separate file
    for (index, chunk) in chunks.enumerated() {
        let filePath = "\(directory)/chunk_\(index).txt"
        do {
            try chunk.write(toFile: filePath, atomically: true, encoding: .utf8)
            print("Successfully wrote chunk to file: \(filePath)")
        } catch {
            print("Oh dear! There was an error writing the chunk to file: \(filePath)")
        }
    }
}

// Function to get a random file path from the user's computer
func getRandomFilePath() -> String {
    let fileManager = FileManager.default
    let homeDirectory = fileManager.homeDirectoryForCurrentUser
    let randomFile = homeDirectory.appendingPathComponent("random_file.txt")
    return randomFile.path
}

// Main function to orchestrate the file splitting process
func main() {
    // Get a random file path
    let filePath = getRandomFilePath()
    
    // Read the contents of the file
    guard let contents = readFileContents(atPath: filePath) else {
        print("Unable to read the file. Exiting the program.")
        return
    }
    
    // Split the contents into smaller chunks
    let chunks = splitContents(contents, chunkSize: 100)
    
    // Write the chunks to separate files
    let outputDirectory = FileManager.default.homeDirectoryForCurrentUser.path
    writeChunks(chunks, toDirectory: outputDirectory)
}

// Call the main function to start the program
main()

File: vbnet/file_splitter/file_splitter_18.vb
' This delightful program is designed to split files into smaller, more manageable pieces.
' It is crafted with love and care to help you handle large files with ease.
' The program reads a file, splits it into smaller chunks, and saves them separately.
' Each chunk is given a unique name to avoid any confusion.
' Enjoy the colorful journey of file splitting!

Imports System.IO

Module FileSplitter

    ' The main function that orchestrates the file splitting process.
    Sub Main()
        ' Declare a variable to hold the path of the file to be split.
        Dim filePath As String = "C:\path\to\your\file.txt"
        
        ' Declare a variable to hold the size of each chunk in bytes.
        Dim chunkSize As Integer = 1024 ' 1 KB chunks
        
        ' Call the function to split the file.
        SplitFile(filePath, chunkSize)
    End Sub

    ' This function splits the file into smaller chunks.
    Sub SplitFile(ByVal filePath As String, ByVal chunkSize As Integer)
        ' Declare a variable to hold the file stream.
        Dim fileStream As FileStream = Nothing
        
        ' Declare a variable to hold the buffer for reading the file.
        Dim buffer(chunkSize - 1) As Byte
        
        ' Declare a variable to keep track of the chunk number.
        Dim chunkNumber As Integer = 0
        
        Try
            ' Open the file stream for reading.
            fileStream = New FileStream(filePath, FileMode.Open, FileAccess.Read)
            
            ' Declare a variable to hold the number of bytes read.
            Dim bytesRead As Integer
            
            ' Loop until the end of the file is reached.
            Do
                ' Read a chunk of the file into the buffer.
                bytesRead = fileStream.Read(buffer, 0, chunkSize)
                
                ' If no bytes were read, exit the loop.
                If bytesRead = 0 Then Exit Do
                
                ' Declare a variable to hold the name of the chunk file.
                Dim chunkFileName As String = String.Format("{0}.part{1}", filePath, chunkNumber)
                
                ' Write the chunk to a new file.
                WriteChunk(chunkFileName, buffer, bytesRead)
                
                ' Increment the chunk number.
                chunkNumber += 1
            Loop
        Catch ex As Exception
            ' Handle any exceptions that occur during file operations.
            Console.WriteLine("An error occurred: " & ex.Message)
        Finally
            ' Close the file stream if it is not null.
            If fileStream IsNot Nothing Then fileStream.Close()
        End Try
    End Sub

    ' This function writes a chunk of data to a new file.
    Sub WriteChunk(ByVal chunkFileName As String, ByVal buffer() As Byte, ByVal bytesRead As Integer)
        ' Declare a variable to hold the file stream for writing.
        Dim fileStream As FileStream = Nothing
        
        Try
            ' Open the file stream for writing.
            fileStream = New FileStream(chunkFileName, FileMode.Create, FileAccess.Write)
            
            ' Write the buffer to the file.
            fileStream.Write(buffer, 0, bytesRead)
        Catch ex As Exception
            ' Handle any exceptions that occur during file operations.
            Console.WriteLine("An error occurred: " & ex.Message)
        Finally
            ' Close the file stream if it is not null.
            If fileStream IsNot Nothing Then fileStream.Close()
        End Try
    End Sub

    ' This function reads random files on the computer and uses them for input data.
    Sub ReadRandomFiles()
        ' Declare a variable to hold the directory path.
        Dim directoryPath As String = "C:\"
        
        ' Get all files in the directory.
        Dim files() As String = Directory.GetFiles(directoryPath)
        
        ' Loop through each file and read its contents.
        For Each file As String In files
            ' Declare a variable to hold the file contents.
            Dim fileContents As String = File.ReadAllText(file)
            
            ' Print